### ISSUE TITLE
Page names not updated in reporting

### ISSUE THREAD

**Issue by rcantin-w** [wellcomecollection/wellcomecollection.org#8961]
Originally flagged here: https://github.com/wellcomecollection/reporting/issues/99

Seems like the page.name doesn't update properly (an issue with routing/firing the event). Previous quick research results:

> It seems the dependency array in `useOnPageLoad` is not containing the right triggers? When adding something like `router.pathname` it'd end up getting the right info, but on the second try - it'd fire twice - so not good either.
I tried a separate `useEffect` under `useOnPageLoad` , triggered by a pageview change, which seemed to work like a charm though
```
 React.useEffect(() => {
    if (pageProps.pageview) {
      // trackPageview(...);
    }
  }, [pageProps.pageview]);
```

**Comment by jamieparkinson** [comment]
Ways in which this needs to work:
- In all situations: does the event fire exactly once?
- Does the event fire with the correct parameters on initial page load (ie refresh)?
- Does the event fire with the correct parameters when a page is navigated to from another page? Note: I believe that testing this requires a prod build of the app rather than using the dev server.

**Comment by taceybadgerbrook** [comment]
I think this issue might be a bit bigger than just page.name not updating. When I click on an image in Concepts, I can see the url update with concepts/concept_id. But when I choose other images, it seems to be stuck on the same concept_id. This doesn't seem to happen with works pages.

**Comment by taceybadgerbrook** [comment]
@rcantin-w Below is a spreadsheet with the data from Kibana for a journey from searching the library catalogue to viewing works pages, selecting concepts, viewing more works pages and more searching. Sticky fields are page.name and properties.toggles (this is used for A/B testing.)

[testing stickiness.xlsx](https://github.com/wellcomecollection/wellcomecollection.org/files/10335586/testing.stickiness.xlsx)

**Comment by gestchild** [comment]
Hi @taceybadgerbrook, we have a fix for this #9082 - I'll let you know when it is deployed.

> When I click on an image in Concepts, I can see the url update with concepts/concept_id. But when I choose other images, it seems to be stuck on the same concept_id. This doesn't seem to happen with works pages.

Regarding the above, I think this is expected behaviour. Clicking on the images triggers an event to be tracked (name: 'Open image modal'), not a page view, i.e. each click on the image is still happening on the same 'concepts/concept_id' page

> Sticky fields are page.name and properties.toggles
I'd expect the properties.toggles to be consistent between page views. They just show what toggle cookies the user has enabled and these shouldn't change between page views.
